### PR TITLE
Update Kotlin compiler and KGP to 1.9.20-RC2

### PR DESCRIPTION
**Pull request by vmishenev**
The current changes are related to https://github.com/JetBrains/kotlin/commit/d797505f06d640b666829bbfb4b7d7c67f812026

### CODE REVIEW

File: core/test-api/src/main/kotlin/testApi/testRunner/TestRunner.kt [hunk 1]
```diff
@@ -191,12 +191,11 @@ public abstract class AbstractTest<M : TestMethods, T : TestBuilder<M>, D : Dokk
...
-        // TODO: feels hacky, find a better way to do it
-        ClassLoader.getSystemResource("kotlin/UInt.kotlin_metadata")
-            ?.file
-            ?.replace("file:", "")
-            ?.replaceAfter(".jar", "")
+    protected val commonStdlibPath: String?  by lazy {
+        // `kotlin-stdlib-common` is legacy
+        // we can use any platform dependency
+        // since common code should be resolved with all platform
+        jvmStdlibPath
```
Comment (IgnatBeresnev) [inline core/test-api/src/main/kotlin/testApi/testRunner/TestRunner.kt:198]: Can we remove/deprecate the existing paths (`jvmStdlibPath`, `jsStdlibPath`, `commonStdlibPath`), and just have a single `stdlibPath` instead?

**Comment by IgnatBeresnev** [comment]
> The current changes are related to https://github.com/JetBrains/kotlin/commit/d797505f06d640b666829bbfb4b7d7c67f812026

Do you know if this has any user-visible effect? That is, can its removal lead to some bugs, inconsistencies or some other problems in the generated documentation/signatures?

File: core/test-api/src/main/kotlin/testApi/testRunner/TestRunner.kt (hunk 1 above)
Comment (vmishenev) [reply core/test-api/src/main/kotlin/testApi/testRunner/TestRunner.kt:198]: No, `jvmStdlibPath` and `jsStdlibPath` contain platform-specific declarations.

**Comment by vmishenev** [comment]
> Do you know if this has any user-visible effect? That is, can its removal lead to some bugs, inconsistencies or some other problems in the generated documentation/signatures?

According to [KT-59643](https://youtrack.jetbrains.com/issue/KT-59643) it disables merging a forward declaration with an actual declaration. It is how the compiler works now. I am unsure it changes the view of signatures.

File: core/test-api/src/main/kotlin/testApi/testRunner/TestRunner.kt (hunk 1 above)
Comment (IgnatBeresnev) [reply core/test-api/src/main/kotlin/testApi/testRunner/TestRunner.kt:198]: Oh, I see what you mean. I misunderstood the comment as "you can use any platform stdlib in any scenario (even use jvm stdlib in js)", but you meant it as "any platform stdlib can be used as a 'common' stdlib"

**Review by IgnatBeresnev — APPROVED** [review]
Could you please update `TestedVersions` just in case?

**Review by IgnatBeresnev — APPROVED** [review]

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [versions]
 
-gradlePlugin-kotlin = "1.9.10"
+gradlePlugin-kotlin = "1.9.20-RC2"
```
Comment (whyoleg) [inline gradle/libs.versions.toml:3]: kotlin 1.9.20 is released already :)

**Pull request merged**